crates/autopilot/src/run_loop.rs

```diff
@@ -405,20 +409,62 @@ impl RunLoop {
 
     /// Execute the solver's solution. Returns Ok when the corresponding
     /// transaction has been mined.
-    async fn settle(&self, driver: &infra::Driver, solved: &Solution) -> Result<(), SettleError> {
+    async fn settle(
+        &self,
+        driver: infra::Driver,
+        solved: &Solution,
+        auction_id: i64,
+    ) -> Result<(), SettleError> {
         let order_ids = solved.order_ids().copied().collect();
         self.persistence
             .store_order_events(order_ids, OrderEventLabel::Executing);
 
+        let eth = self.eth.clone();
+        let persistence = self.persistence.clone();
+        let max_blocks_wait = self.max_blocks_wait;
+        let (cancellation_tx, cancellation_rx) = watch::channel(false);
```

Comment:
Unable to use `oneshot` here since the receiver is used in a loop later.

Reply:
Why do you even need to use channels at all?
It should be possible to not `tokio::spawn` any additional tasks and use something like `futures::future::select()` on the tasks to pick the result of whichever task finishes first.
Also that way I think you don't really have to worry about cancellations.

Reply:
> Also that way I think you don't really have to worry about cancellations.

I didn't quite get it. If the driver returns a result, `wait_for_settlement_transaction` will continue querying db and blockchain in the background until the deadline is reached. I just wanted to avoid that.